crates/query-engine/translation/tests/snapshots/tests__sorting_by_nested_relationship_count.snap

```diff
@@ -3,42 +3,47 @@ source: crates/query-engine/translation/tests/tests.rs
 expression: result
 ---
 SELECT
-  coalesce(json_agg(row_to_json("%3_universe")), '[]') AS "universe"
+  coalesce(json_agg(row_to_json("%4_universe")), '[]') AS "universe"
 FROM
   (
     SELECT
       *
     FROM
       (
         SELECT
-          coalesce(json_agg(row_to_json("%4_rows")), '[]') AS "rows"
+          coalesce(json_agg(row_to_json("%5_rows")), '[]') AS "rows"
         FROM
           (
             SELECT
-              "%0_Artist"."Name" AS "Name"
+              "%3_Artist"."Name" AS "Name"
             FROM
-              "public"."Artist" AS "%0_Artist"
-              LEFT OUTER JOIN LATERAL (
+              (
                 SELECT
-                  count("%1_ORDER_PART_Album"."AlbumId") AS "AlbumId"
+                  "%0_Artist".*
                 FROM
-                  (
+                  "public"."Artist" AS "%0_Artist"
+                  LEFT OUTER JOIN LATERAL (
                     SELECT
-                      "%1_ORDER_PART_Album"."AlbumId" AS "AlbumId"
+                      count("%1_ORDER_PART_Album"."AlbumId") AS "AlbumId"
                     FROM
-                      "public"."Album" AS "%1_ORDER_PART_Album"
-                    WHERE
                       (
-                        "%0_Artist"."ArtistId" = "%1_ORDER_PART_Album"."ArtistId"
-                      )
-                  ) AS "%1_ORDER_PART_Album"
-              ) AS "%2_ORDER_FOR_Artist" ON ('true')
-            ORDER BY
-              "%2_ORDER_FOR_Artist"."AlbumId" DESC
-            LIMIT
-              3
-          ) AS "%4_rows"
-      ) AS "%4_rows"
-  ) AS "%3_universe";
+                        SELECT
+                          "%1_ORDER_PART_Album"."AlbumId" AS "AlbumId"
+                        FROM
+                          "public"."Album" AS "%1_ORDER_PART_Album"
+                        WHERE
+                          (
+                            "%0_Artist"."ArtistId" = "%1_ORDER_PART_Album"."ArtistId"
+                          )
+                      ) AS "%1_ORDER_PART_Album"
+                  ) AS "%2_ORDER_FOR_Artist" ON ('true')
+                ORDER BY
+                  "%2_ORDER_FOR_Artist"."AlbumId" DESC
```

Comment:
V2 also has a copy of the order by clause on the parent subquery to this one.
I presume there's the potential for ordering to be lost when joining with other tables?

Are we comfortable without adding that additional clause?

Reply:
I would feel a lot more comfortable if we added a test so we know we're not creating a regression.

Reply:
Looks like we have a test for ordering across relationships already, so this is fine IMO. https://github.com/hasura/ndc-postgres/blob/210ea0510a8621c32e515d017d0a44272f33de78/crates/tests/tests-common/goldenfiles/sorting_by_nested_relationship_column_with_predicate.json#L4

Reply:
By all means open a ticket for further investigation, I am satisfied for the moment though.